midas.py

```diff
@@ -0,0 +1,166 @@
+import db_handler
+from terrorist import Terrorist
+from organization import Organization
+from event import Event
+from sqlalchemy import MetaData
+meta = MetaData()
+
+
+def add_terrorist(session, name, last_name, role, location):
+    terrorist_instance = Terrorist(name=name, last_name=last_name, role=role, location=location)
+    return add_entity(session, terrorist_instance)
+
+
+def get_terrorist(session, entity_id):
+    return get_entity_by_id(session, entity_id, Terrorist)
+
+
+def add_organization(session, prime_location, name):
+    organization_instance = Organization(prime_location=prime_location, name=name)
+    return add_entity(session, organization_instance)
+
+
+def get_organization(session, entity_id):
+    return get_entity_by_id(session, entity_id, Organization)
+
+
+def add_event(session, location, date):
+    event_instance = Event(location=location, date=date)
+    return add_entity(session, event_instance)
+
+
+def get_event(session, entity_id):
+    return get_entity_by_id(session, entity_id, Event)
+
+
+def add_entity(session, entity):
+    """
+    Persisting an entity to thje DB. Used as a base by all entity types
+    :param session:
+    :param entity:
+    :return:
+    """
+    session.add(entity)
+    session.commit()
+    return entity
+
+
+def get_entity_by_id(session, entity_id, entity_type):
+    results = session.query(entity_type).filter(entity_type.id == entity_id)
+    return results[0] if results.count() > 0 else None
+
+
+def add_member_to_organization(session, terrorist_id, organization_id):
+    """
+    Link a terrorist to an organization
+
+    :param session:
+    :param terrorist_id:
+    :param organization_id:
+    :return:
+    """
+    matched_organization = get_entity_by_id(session, organization_id, Organization)
+    matched_terrorist = get_entity_by_id(session, terrorist_id, Terrorist)
+    matched_organization.members.append(matched_terrorist)
+    session.commit()
+
+
+def add_member_to_event(session, terrorist_id, event_id):
+    """
+    Link a member to an event
+    :param session:
+    :param terrorist_id:
+    :param event_id:
+    :return:
+    """
+    matched_event = get_entity_by_id(session, event_id, Event)
+    matched_terrorist = get_entity_by_id(session, terrorist_id, Terrorist)
+    matched_event.participants.append(matched_terrorist)
+    session.commit()
+
+
+def get_members_not_kalab(session):
+    """
+    Get a list of members which are not stationed in the original location
+    :param session:
+    :return:
+    """
+    all_terrorists = Terrorist.get(session).all()
+    return [t.id for t in all_terrorists if not _is_member_kalab(t)]
```

Comment:
You should use SQLAlchemy to filter the results.

Reply:
Done